dependencies.go

```diff
@@ -78,20 +82,25 @@ type faultyDiskDependency struct {
 	// failDenominator, and it starts at 2. This means that the more calls to
 	// WriteAt, the less likely the write is to fail. All calls will start
 	// automatically failing after writeLimit writes.
-	failDenominator int
-	writeLimit      int
-	failed          bool
+	failDenominator *uint64
+	writeLimit      *uint64
+	failed          *bool
+	disabled        *bool
 	mu              *sync.Mutex
```

Comment:
What is the reason for making these fields pointers?

Reply:
The interface for `Create` and other functions need the dependency to be a value object instead of a pointer. That means the contents of the value objects need to be pointer to be able to pass the dependency to multiple functions. Otherwise each function will have its own instance of the dependency.

Reply:
You just have to make the methods act on a pointer, I'll make the PR for that.